<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content />


<title>Using-SIMMS.utf8</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>







<p><style type="text/css">body {
font-family: Helvetica, arial, sans-serif;
font-size: 14px;
line-height: 1.6;
padding-top: 10px;
padding-bottom: 10px;
background-color: white;
padding: 30px; }
body > *:first-child {
margin-top: 0 !important; }
body > *:last-child {
margin-bottom: 0 !important; }
a {
color: #4183C4; }
a.absent {
color: #cc0000; }
a.anchor {
display: block;
padding-left: 30px;
margin-left: -30px;
cursor: pointer;
position: absolute;
top: 0;
left: 0;
bottom: 0; }
h1, h2, h3, h4, h5, h6 {
margin: 20px 0 10px;
padding: 0;
font-weight: bold;
-webkit-font-smoothing: antialiased;
cursor: text;
position: relative; }
h1:hover a.anchor, h2:hover a.anchor, h3:hover a.anchor, h4:hover a.anchor, h5:hover a.anchor, h6:hover a.anchor {
background: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAA09pVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMy1jMDExIDY2LjE0NTY2MSwgMjAxMi8wMi8wNi0xNDo1NjoyNyAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvIiB4bWxuczp4bXBNTT0iaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wL21tLyIgeG1sbnM6c3RSZWY9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9zVHlwZS9SZXNvdXJjZVJlZiMiIHhtcDpDcmVhdG9yVG9vbD0iQWRvYmUgUGhvdG9zaG9wIENTNiAoMTMuMCAyMDEyMDMwNS5tLjQxNSAyMDEyLzAzLzA1OjIxOjAwOjAwKSAgKE1hY2ludG9zaCkiIHhtcE1NOkluc3RhbmNlSUQ9InhtcC5paWQ6OUM2NjlDQjI4ODBGMTFFMTg1ODlEODNERDJBRjUwQTQiIHhtcE1NOkRvY3VtZW50SUQ9InhtcC5kaWQ6OUM2NjlDQjM4ODBGMTFFMTg1ODlEODNERDJBRjUwQTQiPiA8eG1wTU06RGVyaXZlZEZyb20gc3RSZWY6aW5zdGFuY2VJRD0ieG1wLmlpZDo5QzY2OUNCMDg4MEYxMUUxODU4OUQ4M0REMkFGNTBBNCIgc3RSZWY6ZG9jdW1lbnRJRD0ieG1wLmRpZDo5QzY2OUNCMTg4MEYxMUUxODU4OUQ4M0REMkFGNTBBNCIvPiA8L3JkZjpEZXNjcmlwdGlvbj4gPC9yZGY6UkRGPiA8L3g6eG1wbWV0YT4gPD94cGFja2V0IGVuZD0iciI/PsQhXeAAAABfSURBVHjaYvz//z8DJYCRUgMYQAbAMBQIAvEqkBQWXI6sHqwHiwG70TTBxGaiWwjCTGgOUgJiF1J8wMRAIUA34B4Q76HUBelAfJYSA0CuMIEaRP8wGIkGMA54bgQIMACAmkXJi0hKJQAAAABJRU5ErkJggg==) no-repeat 10px center;
text-decoration: none; }
h1 tt, h1 code {
font-size: inherit; }
h2 tt, h2 code {
font-size: inherit; }
h3 tt, h3 code {
font-size: inherit; }
h4 tt, h4 code {
font-size: inherit; }
h5 tt, h5 code {
font-size: inherit; }
h6 tt, h6 code {
font-size: inherit; }
h1 {
font-size: 28px;
color: black; }
h2 {
font-size: 24px;
border-bottom: 1px solid #cccccc;
color: black; }
h3 {
font-size: 18px; }
h4 {
font-size: 16px; }
h5 {
font-size: 14px; }
h6 {
color: #777777;
font-size: 14px; }
p, blockquote, ul, ol, dl, li, table, pre {
margin: 15px 0; }
hr {
background: transparent url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAYAAAAECAYAAACtBE5DAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAyJpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADw/eHBhY2tldCBiZWdpbj0i77u/IiBpZD0iVzVNME1wQ2VoaUh6cmVTek5UY3prYzlkIj8+IDx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IkFkb2JlIFhNUCBDb3JlIDUuMC1jMDYwIDYxLjEzNDc3NywgMjAxMC8wMi8xMi0xNzozMjowMCAgICAgICAgIj4gPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4gPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIgeG1sbnM6eG1wPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvIiB4bWxuczp4bXBNTT0iaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wL21tLyIgeG1sbnM6c3RSZWY9Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9zVHlwZS9SZXNvdXJjZVJlZiMiIHhtcDpDcmVhdG9yVG9vbD0iQWRvYmUgUGhvdG9zaG9wIENTNSBNYWNpbnRvc2giIHhtcE1NOkluc3RhbmNlSUQ9InhtcC5paWQ6OENDRjNBN0E2NTZBMTFFMEI3QjRBODM4NzJDMjlGNDgiIHhtcE1NOkRvY3VtZW50SUQ9InhtcC5kaWQ6OENDRjNBN0I2NTZBMTFFMEI3QjRBODM4NzJDMjlGNDgiPiA8eG1wTU06RGVyaXZlZEZyb20gc3RSZWY6aW5zdGFuY2VJRD0ieG1wLmlpZDo4Q0NGM0E3ODY1NkExMUUwQjdCNEE4Mzg3MkMyOUY0OCIgc3RSZWY6ZG9jdW1lbnRJRD0ieG1wLmRpZDo4Q0NGM0E3OTY1NkExMUUwQjdCNEE4Mzg3MkMyOUY0OCIvPiA8L3JkZjpEZXNjcmlwdGlvbj4gPC9yZGY6UkRGPiA8L3g6eG1wbWV0YT4gPD94cGFja2V0IGVuZD0iciI/PqqezsUAAAAfSURBVHjaYmRABcYwBiM2QSA4y4hNEKYDQxAEAAIMAHNGAzhkPOlYAAAAAElFTkSuQmCC) repeat-x 0 0;
border: 0 none;
color: #cccccc;
height: 4px;
padding: 0;
}
body > h2:first-child {
margin-top: 0;
padding-top: 0; }
body > h1:first-child {
margin-top: 0;
padding-top: 0; }
body > h1:first-child + h2 {
margin-top: 0;
padding-top: 0; }
body > h3:first-child, body > h4:first-child, body > h5:first-child, body > h6:first-child {
margin-top: 0;
padding-top: 0; }
a:first-child h1, a:first-child h2, a:first-child h3, a:first-child h4, a:first-child h5, a:first-child h6 {
margin-top: 0;
padding-top: 0; }
h1 p, h2 p, h3 p, h4 p, h5 p, h6 p {
margin-top: 0; }
li p.first {
display: inline-block; }
li {
margin: 0; }
ul, ol {
padding-left: 30px; }
ul :first-child, ol :first-child {
margin-top: 0; }
dl {
padding: 0; }
dl dt {
font-size: 14px;
font-weight: bold;
font-style: italic;
padding: 0;
margin: 15px 0 5px; }
dl dt:first-child {
padding: 0; }
dl dt > :first-child {
margin-top: 0; }
dl dt > :last-child {
margin-bottom: 0; }
dl dd {
margin: 0 0 15px;
padding: 0 15px; }
dl dd > :first-child {
margin-top: 0; }
dl dd > :last-child {
margin-bottom: 0; }
blockquote {
border-left: 4px solid #dddddd;
padding: 0 15px;
color: #777777; }
blockquote > :first-child {
margin-top: 0; }
blockquote > :last-child {
margin-bottom: 0; }
table {
padding: 0;border-collapse: collapse; }
table tr {
border-top: 1px solid #cccccc;
background-color: white;
margin: 0;
padding: 0; }
table tr:nth-child(2n) {
background-color: #f8f8f8; }
table tr th {
font-weight: bold;
border: 1px solid #cccccc;
margin: 0;
padding: 6px 13px; }
table tr td {
border: 1px solid #cccccc;
margin: 0;
padding: 6px 13px; }
table tr th :first-child, table tr td :first-child {
margin-top: 0; }
table tr th :last-child, table tr td :last-child {
margin-bottom: 0; }
img {
max-width: 100%; }
span.frame {
display: block;
overflow: hidden; }
span.frame > span {
border: 1px solid #dddddd;
display: block;
float: left;
overflow: hidden;
margin: 13px 0 0;
padding: 7px;
width: auto; }
span.frame span img {
display: block;
float: left; }
span.frame span span {
clear: both;
color: #333333;
display: block;
padding: 5px 0 0; }
span.align-center {
display: block;
overflow: hidden;
clear: both; }
span.align-center > span {
display: block;
overflow: hidden;
margin: 13px auto 0;
text-align: center; }
span.align-center span img {
margin: 0 auto;
text-align: center; }
span.align-right {
display: block;
overflow: hidden;
clear: both; }
span.align-right > span {
display: block;
overflow: hidden;
margin: 13px 0 0;
text-align: right; }
span.align-right span img {
margin: 0;
text-align: right; }
span.float-left {
display: block;
margin-right: 13px;
overflow: hidden;
float: left; }
span.float-left span {
margin: 13px 0 0; }
span.float-right {
display: block;
margin-left: 13px;
overflow: hidden;
float: right; }
span.float-right > span {
display: block;
overflow: hidden;
margin: 13px auto 0;
text-align: right; }
code, tt {
margin: 0 2px;
padding: 0 5px;
white-space: nowrap;
border: 1px solid #eaeaea;
background-color: #f8f8f8;
border-radius: 3px; }
pre code {
margin: 0;
padding: 0;
white-space: pre;
border: none;
background: transparent; }
.highlight pre {
background-color: #f8f8f8;
border: 1px solid #cccccc;
font-size: 13px;
line-height: 19px;
overflow: auto;
padding: 6px 10px;
border-radius: 3px; }
pre {
background-color: #f8f8f8;
border: 1px solid #cccccc;
font-size: 13px;
line-height: 19px;
overflow: auto;
padding: 6px 10px;
border-radius: 3px; }
pre code, pre tt {
background-color: transparent;
border: none; }
sup {
font-size: 0.83em;
vertical-align: super;
line-height: 0;
}
* {
-webkit-print-color-adjust: exact;
}
@media screen and (min-width: 914px) {
body {
width: 854px;
margin:0 auto;
}
}
@media print {
table, pre {
page-break-inside: avoid;
}
pre {
word-wrap: break-word;
}
}</style></p>
<div id="subnetwork-integration-for-multi-modal-signatures-simms" class="section level2">
<h2>Subnetwork Integration for Multi-modal Signatures (SIMMS)</h2>
<div id="authors-syed-haider-paul-c.-boutros" class="section level3">
<h3>Author(s): Syed Haider, Paul C. Boutros</h3>
</div>
<div id="maintainer-syed-haider-syed.haidericr.ac.uk" class="section level3">
<h3>Maintainer: Syed Haider (<a href="mailto:Syed.Haider@icr.ac.uk" class="email">Syed.Haider@icr.ac.uk</a>)</h3>
</div>
<div id="dated-2021-01-07" class="section level3">
<h3>Dated: 2021-01-07</h3>
</div>
</div>
<div id="table-of-contents" class="section level2">
<h2>Table of Contents</h2>
<ul>
<li><a href="#Overview">Overview</a></li>
<li><a href="#Networks-database">Networks database</a></li>
<li><a href="#Molecular-profiles-and-clinical-data">Molecular profiles and clinical data</a></li>
<li><a href="#Example-R-script">Example R-script</a></li>
<li><a href="#Output">Output</a></li>
</ul>
</div>
<div id="overview" class="section level2">
<h2><a id="Overview"></a> Overview</h2>
<p>SIMMS enables intergration of molecular profiles with functional networks such as protein-protein interaction networks. This document shows how to use SIMMS from a simple use case of integrating mRNA abundance data with pathways derived protein-protein networks, through to more sophisticated use cases of integrating multiple molecular profiles such as mRNA abundance, DNA copy-number and DNA-methylation data. Note, SIMMS requires patient outcome data as dependent variable. In current implementation, it works with survival data <code>Survival time, Survival status</code>. Please setup the dataset metadata directories before using SIMMS.</p>
<p>There are a couple of directories that one needs to setup and ensure the contents are in correct format. These directories contain:</p>
<ul>
<li>Networks database (optional)</li>
<li>Molecular profiles and clinical data</li>
</ul>
<p>The structure and format of these inputs is described below.</p>
</div>
<div id="networks-database" class="section level2">
<h2><a id="Networks-database"></a> Networks database</h2>
<p>SIMMS rely on functionally or computationally derived networks in order to meaningfully integrate molecular profiles. By default, SIMMS package has a built in database of pathway derived protein-protein interaction networks extracted from (a dated version) Reactome, BioCarta and NCI-PID. This database can be accessed through various SIMMS functions using the parameter setting <code>networks.database = &quot;default&quot;</code>. If you would like to create your own networks database for subsequent use with SIMMS, please note that the networks database is organised in two files:</p>
<pre><code>pathway_based_sub_networks.txt
pathway_based_networks_flattened.txt</code></pre>
<p>File <code>pathway_based_sub_networks.txt</code> contains all the subnetworks. For instance a hypothetical subnetwork of five genes <code>ERBB2, EGFR, MKI67, ESR1, PGR</code> with four interactions <code>(PGR-ESR1, EGFR-ERBB2, EGFR-PGR, MKI67-ESR1)</code>, and another hypothetical subnetwork of three genes PDK1, AKT3 and PIK3CA with two interactions <code>(PDK1-PIK3CA, AKT3-PIK3CA)</code> using Entrez Gene IDs would look like (tab-delimited):</p>
<pre><code>#ID=0001    NAME=Module.1
ID=0001 5241    2099
ID=0001 1956    2064
ID=0001 1956    5241
ID=0001 4288    2099
#ID=0002    NAME=Module.2
ID=0002 5163    5290
ID=0002 10000   5290</code></pre>
<p>Please note that the only Entrez IDs are supported in this file, which should match the Entrez IDs format in the molecular data files (except additional _at suffix in the molecular data files).</p>
<p>File <code>pathway_based_networks_flattened.txt</code> is a non-redundant, without self-interactions, quick lookup table of all the pairwise interactions present in the file <code>pathway_based_sub_networks.txt</code>. This should not contain the subnetwork name and ID annotations. For example, the contents of this file for the above-mentioned subnetworks would look like (tab-delimited):</p>
<pre><code>GeneID1 GeneID2
5241    2099
1956    2064
1956    5241
4288    2099
5163    5290
10000   5290    </code></pre>
<p>Please note that only Entrez IDs are supported in this file, which should match the Entrez IDs format in the molecular data files (except additional _at suffix in the molecular data files). If you have a gene list without known interactions, and would like to try Node-only model (model 2), you can create a subnetwork with random interactions in <code>pathway_based_sub_networks.txt</code> as long as all the features (genes) are present and connected. Same goes for <code>pathway_based_networks_flattened.txt</code>.</p>
<p>The two database files are placed inside a directory you wish to call your database. For example: <code>MyNetworkDB</code>, and this directory should be placed under:</p>
<blockquote>
<p>SIMMS/inst/programdata/networkdb/</p>
</blockquote>
<p>You would need to build and install SIMMS package again and your networks database should be available to use through relevant SIMMS’ functions with parameter setting <code>networks.database = &quot;MyNetworkDB&quot;</code></p>
</div>
<div id="molecular-profiles-and-clinical-data" class="section level2">
<h2><a id="Molecular-profiles-and-clinical-data"></a> Molecular profiles and clinical data</h2>
<p>The main input to SIMMS is molecular and clinical data. Currently tested datatypes are mRNA abundance, DNA copy-number and DNA methylation datasets. mRNA abundance and DNA methylation profiles are expected to be in continuous scale, while DNA copy-number profiles are expected to be gene copy-number calls <code>(-2,-1,0,1,2) or (-1,0,1)</code> representing deletions (-ve), neutral (0), and gains/amplifications (+ve) calls. Given that SIMMS support both continuous and categorical/ordinal profiles, any datatype can be used as input. The table below shows the supported/unsupported datatypes and respective examples:</p>
<!-- html table generated in R 3.6.1 by xtable 1.8-4 package -->
<!-- Thu Jan  7 14:05:58 2021 -->
<table border="1">
<tr>
<th>
Data type
</th>
<th>
Example data
</th>
<th>
Example profile
</th>
</tr>
<tr>
<td>
{ <em>x ∈ ℝ</em> : <em>x ≥ 0</em> }
</td>
<td>
0, 2, 1.37, 7.04, 9.68
</td>
<td>
log2 mRNA or miRNA abundance
</td>
</tr>
<tr>
<td>
{ <em>x ∈ ℝ</em> : <em>x ≥ 0</em> }
</td>
<td>
0.1, 0.38, 0.78, 0.22, 0.98
</td>
<td>
DNA methylation beta values
</td>
</tr>
<tr>
<td>
{ <em>x ∈ ℤ</em> }
</td>
<td>
-2, -1, 0, 1, 2
</td>
<td>
copy-number calls. Reference group = 0 (Neutral/Diploid)
</td>
</tr>
<tr>
<td>
{ <em>x ∈ ℤ</em> : <em>x ≥ 0</em> }
</td>
<td>
0, 1, 2, 3
</td>
<td>
mutation data. Reference group = 0 (Wildtype)
</td>
</tr>
<tr>
<td>
{ <em>x ∈ ℤ</em> : <em>x ≠ 0</em> }
</td>
<td>
-2, -1, 1, 2
</td>
<td>
Unsupported due to missing reference group 0
</td>
</tr>
<tr>
<td>
{ <em>x ∉ ℝ</em> }
</td>
<td>
WT, Mutant, Gain, Deleted
</td>
<td>
Unsupported due to alphabets
</td>
</tr>
</table>
<p>Molecular profiles are strictly tab-delimited, and are expected to be in feature (gene) by sample (patient) matrices. Genes should be represented using Entrez IDs followed by _at suffix. For mRNA abundance data, if you have pre-processed your data with <a href="http://brainarray.mbni.med.umich.edu/brainarray/Database/CustomCDF/genomic_curated_CDF.asp">BrainArray Entrez CDFs</a> Entrez CDF, your data should be SIMMS compatible by default. Otherwise, please map your dataset features e.g genes to Entrez IDs followed by _at suffix (e.g 5290_at).</p>
<p>Clinical profiles are strictly tab-delimited, sample (patient) by annotation matrices. The annotation columns must contain survival columns <code>Survival time, Survival status, Survival time unit</code>. The row names should match the column names in the molecular profiles’ matrices. These columns in clinical annotation file could be called anything as long as these are correctly identified through the metadata file described below.</p>
<p>Both molecular and clinical annotations are read by SIMMS through a tab-delimited metadata file called <code>datasets.txt</code>. An example of the contents of this file are shown below:</p>
<pre><code>dataset mRNA    cna methylation annotations survstat    survtime    survtime.unit
Breastdata1 mRNA_abundance.txt  CNA.txt DNA_methylation.txt patient_annotation.txt  e.os    t.os    t.os.unit
Breastdata2 mRNA_abundance.txt  CNA.txt DNA_methylation.txt patient_annotation.txt  e.os    t.os    t.os.unit</code></pre>
<p>Here, each row contain an entry for a dataset e.g Breastdata1 and Breastdata2, each having its own directory on the filesystem. mRNA, cna and methylation columns contain the file names of each of the these datatypes. annotations column contains the names of the annotation files for each dataset. All the datatypes and annotation files for a given dataset must be inside the dataset directory (e.g Breastdata1/). survstat, survtime, and survtime.unit columns contain the column names of <code>Survival status, Survival time and Survival time unit</code>, respectively. These column names are expected to match the column names in the annotations files. annotation data file must also have a column <code>Tumour</code> with possible values of <code>Yes|No</code>. All analyses will be limited to those samples with <code>Yes</code> in the <code>Tumour</code> column. Further subsetting of molecular and annotation datasets is enable using parameter <code>subset</code> in SIMMS’ functions <code>?derive.network.features</code> &amp; <code>?prepare.training.validation.datasets</code>. For convenience sake, an example test dataset <code>testdata</code> containing metadata file <code>datasets.txt</code>, mRNA abundance profiles and respective clinical data is bundled with SIMMS package, and can be found under:</p>
<blockquote>
<p>SIMMS/inst/programdata/testdata/</p>
</blockquote>
<p>There is no need to drop your datasets inside the package, or need to rebuild the package. You can just point to your datasets through SIMMS package keeping them anywhere on your filesystem.</p>
</div>
<div id="example-r-script" class="section level2">
<h2><a id="Example-R-script"></a> Example R-script</h2>
<p>Lets start with a simple case. We have two mRNA abundance datasets; <code>Breastdata1, Breastdata2</code>. We would like to identify prognostic biomarkers using Breastdata1 and validate on Breastdata2. Assuming these two datasets are setup correctly (as described in the previous section), a typical workflow would be:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a><span class="kw">options</span>(<span class="st">&quot;warn&quot;</span> =<span class="st"> </span><span class="dv">-1</span>);</span>
<span id="cb5-2"><a href="#cb5-2"></a></span>
<span id="cb5-3"><a href="#cb5-3"></a><span class="co"># load SIMMS library</span></span>
<span id="cb5-4"><a href="#cb5-4"></a><span class="kw">library</span>(<span class="st">&quot;SIMMS&quot;</span>);</span>
<span id="cb5-5"><a href="#cb5-5"></a></span>
<span id="cb5-6"><a href="#cb5-6"></a><span class="co"># path of the data directory containing Breastdata1/ and Breastdata2/ subdirectories</span></span>
<span id="cb5-7"><a href="#cb5-7"></a>data.directory &lt;-<span class="st"> </span>SIMMS<span class="op">::</span><span class="kw">get.program.defaults</span>(<span class="dt">networks.database =</span> <span class="st">&quot;test&quot;</span>)[[<span class="st">&quot;test.data.dir&quot;</span>]];</span>
<span id="cb5-8"><a href="#cb5-8"></a></span>
<span id="cb5-9"><a href="#cb5-9"></a><span class="co"># path of the directory where results will be stored</span></span>
<span id="cb5-10"><a href="#cb5-10"></a>output.directory &lt;-<span class="st"> </span><span class="kw">tempdir</span>();</span>
<span id="cb5-11"><a href="#cb5-11"></a></span>
<span id="cb5-12"><a href="#cb5-12"></a><span class="co"># molecular profiles to be used</span></span>
<span id="cb5-13"><a href="#cb5-13"></a>data.types &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;mRNA&quot;</span>);</span>
<span id="cb5-14"><a href="#cb5-14"></a></span>
<span id="cb5-15"><a href="#cb5-15"></a><span class="co"># feature selection datasets</span></span>
<span id="cb5-16"><a href="#cb5-16"></a><span class="co"># name of the dataset directory containing mRNA abundance and annotation profiles of training dataset/s</span></span>
<span id="cb5-17"><a href="#cb5-17"></a>feature.selection.datasets &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Breastdata1&quot;</span>);</span>
<span id="cb5-18"><a href="#cb5-18"></a></span>
<span id="cb5-19"><a href="#cb5-19"></a><span class="co"># model training datasets, ideally same as feature selection datasets</span></span>
<span id="cb5-20"><a href="#cb5-20"></a><span class="co"># name of the dataset directory containing mRNA abundance and annotation profiles of training dataset/s</span></span>
<span id="cb5-21"><a href="#cb5-21"></a>training.datasets &lt;-<span class="st"> </span>feature.selection.datasets;</span>
<span id="cb5-22"><a href="#cb5-22"></a></span>
<span id="cb5-23"><a href="#cb5-23"></a><span class="co"># validation datasets</span></span>
<span id="cb5-24"><a href="#cb5-24"></a><span class="co"># name of the dataset directory containing mRNA abundance and annotation profiles of validation dataset/s</span></span>
<span id="cb5-25"><a href="#cb5-25"></a>validation.datasets &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Breastdata2&quot;</span>);</span>
<span id="cb5-26"><a href="#cb5-26"></a></span>
<span id="cb5-27"><a href="#cb5-27"></a><span class="co"># all the possible P value thresholds that one may consider applying to feature selection process.</span></span>
<span id="cb5-28"><a href="#cb5-28"></a><span class="co"># its the P value of univariate (genewise) Cox model statistics</span></span>
<span id="cb5-29"><a href="#cb5-29"></a>feature.selection.p.thresholds &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="fl">0.5</span>);</span>
<span id="cb5-30"><a href="#cb5-30"></a></span>
<span id="cb5-31"><a href="#cb5-31"></a><span class="co"># one of the P values above, to be used for subsequent analysis. Not a vector for performance reasons</span></span>
<span id="cb5-32"><a href="#cb5-32"></a>feature.selection.p.threshold &lt;-<span class="st"> </span><span class="fl">0.5</span>;</span>
<span id="cb5-33"><a href="#cb5-33"></a></span>
<span id="cb5-34"><a href="#cb5-34"></a><span class="co"># names of the learning algorithms to be used for the final multivarite model</span></span>
<span id="cb5-35"><a href="#cb5-35"></a>learning.algorithms &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;backward&quot;</span>, <span class="st">&quot;forward&quot;</span>, <span class="st">&quot;glm&quot;</span>, <span class="st">&quot;randomforest&quot;</span>);</span>
<span id="cb5-36"><a href="#cb5-36"></a></span>
<span id="cb5-37"><a href="#cb5-37"></a><span class="co"># top features to be used for model selection (Backwards elimination, Forward selection, GLM, Random survival forest)</span></span>
<span id="cb5-38"><a href="#cb5-38"></a><span class="co"># you can try a number of different model selection runs by specifying a vector of top n features</span></span>
<span id="cb5-39"><a href="#cb5-39"></a>top.n.features &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">5</span>);</span>
<span id="cb5-40"><a href="#cb5-40"></a></span>
<span id="cb5-41"><a href="#cb5-41"></a><span class="co"># truncate survival</span></span>
<span id="cb5-42"><a href="#cb5-42"></a>truncate.survival &lt;-<span class="st"> </span><span class="dv">10</span>;</span>
<span id="cb5-43"><a href="#cb5-43"></a></span>
<span id="cb5-44"><a href="#cb5-44"></a><span class="co"># calculate per feature univariate coefficients in training sets</span></span>
<span id="cb5-45"><a href="#cb5-45"></a><span class="kw">derive.network.features</span>(</span>
<span id="cb5-46"><a href="#cb5-46"></a>    <span class="dt">data.directory =</span> data.directory,</span>
<span id="cb5-47"><a href="#cb5-47"></a>    <span class="dt">output.directory =</span> output.directory,</span>
<span id="cb5-48"><a href="#cb5-48"></a>    <span class="dt">data.types =</span> data.types,</span>
<span id="cb5-49"><a href="#cb5-49"></a>    <span class="dt">feature.selection.datasets =</span> feature.selection.datasets,</span>
<span id="cb5-50"><a href="#cb5-50"></a>    <span class="dt">feature.selection.p.thresholds =</span> feature.selection.p.thresholds,</span>
<span id="cb5-51"><a href="#cb5-51"></a>    <span class="dt">networks.database =</span> <span class="st">&quot;test&quot;</span>, <span class="co"># or &quot;default&quot; for Reactome/BioCarta/NCI-PID</span></span>
<span id="cb5-52"><a href="#cb5-52"></a>    <span class="dt">truncate.survival =</span> truncate.survival</span>
<span id="cb5-53"><a href="#cb5-53"></a>    );</span>
<span id="cb5-54"><a href="#cb5-54"></a></span>
<span id="cb5-55"><a href="#cb5-55"></a><span class="co"># calculate per-subnetwork scores in both training and validation sets</span></span>
<span id="cb5-56"><a href="#cb5-56"></a><span class="kw">prepare.training.validation.datasets</span>(</span>
<span id="cb5-57"><a href="#cb5-57"></a>    <span class="dt">data.directory =</span> data.directory,</span>
<span id="cb5-58"><a href="#cb5-58"></a>    <span class="dt">output.directory =</span> output.directory,</span>
<span id="cb5-59"><a href="#cb5-59"></a>    <span class="dt">data.types =</span> data.types,</span>
<span id="cb5-60"><a href="#cb5-60"></a>    <span class="dt">p.threshold =</span> feature.selection.p.threshold,</span>
<span id="cb5-61"><a href="#cb5-61"></a>    <span class="dt">feature.selection.datasets =</span> feature.selection.datasets,</span>
<span id="cb5-62"><a href="#cb5-62"></a>    <span class="dt">datasets =</span> <span class="kw">c</span>(training.datasets, validation.datasets),</span>
<span id="cb5-63"><a href="#cb5-63"></a>    <span class="dt">networks.database =</span> <span class="st">&quot;test&quot;</span>, <span class="co"># or &quot;default&quot; for Reactome/BioCarta/NCI-PID</span></span>
<span id="cb5-64"><a href="#cb5-64"></a>    <span class="dt">truncate.survival =</span> truncate.survival</span>
<span id="cb5-65"><a href="#cb5-65"></a>    );</span>
<span id="cb5-66"><a href="#cb5-66"></a></span>
<span id="cb5-67"><a href="#cb5-67"></a><span class="co"># iterate over varying top n features, identify and validate survival models</span></span>
<span id="cb5-68"><a href="#cb5-68"></a><span class="cf">for</span> (top.n <span class="cf">in</span> top.n.features) {</span>
<span id="cb5-69"><a href="#cb5-69"></a></span>
<span id="cb5-70"><a href="#cb5-70"></a>    <span class="co"># create classifier assessing univariate prognostic power of subnetwork modules (Train and Validate)</span></span>
<span id="cb5-71"><a href="#cb5-71"></a>    ret &lt;-<span class="st"> </span><span class="kw">create.classifier.univariate</span>(</span>
<span id="cb5-72"><a href="#cb5-72"></a>        <span class="dt">data.directory =</span> data.directory,</span>
<span id="cb5-73"><a href="#cb5-73"></a>        <span class="dt">output.directory =</span> output.directory,</span>
<span id="cb5-74"><a href="#cb5-74"></a>        <span class="dt">feature.selection.datasets =</span> feature.selection.datasets,</span>
<span id="cb5-75"><a href="#cb5-75"></a>        <span class="dt">feature.selection.p.threshold =</span> feature.selection.p.threshold,</span>
<span id="cb5-76"><a href="#cb5-76"></a>        <span class="dt">training.datasets =</span> training.datasets,</span>
<span id="cb5-77"><a href="#cb5-77"></a>        <span class="dt">validation.datasets =</span> validation.datasets,</span>
<span id="cb5-78"><a href="#cb5-78"></a>        <span class="dt">top.n.features =</span> top.n</span>
<span id="cb5-79"><a href="#cb5-79"></a>        );</span>
<span id="cb5-80"><a href="#cb5-80"></a></span>
<span id="cb5-81"><a href="#cb5-81"></a>    <span class="co"># create a multivariate classifier (Train and Validate)</span></span>
<span id="cb5-82"><a href="#cb5-82"></a>    ret &lt;-<span class="st"> </span><span class="kw">create.classifier.multivariate</span>(</span>
<span id="cb5-83"><a href="#cb5-83"></a>        <span class="dt">data.directory =</span> data.directory,</span>
<span id="cb5-84"><a href="#cb5-84"></a>        <span class="dt">output.directory =</span> output.directory,</span>
<span id="cb5-85"><a href="#cb5-85"></a>        <span class="dt">feature.selection.datasets =</span> feature.selection.datasets,</span>
<span id="cb5-86"><a href="#cb5-86"></a>        <span class="dt">feature.selection.p.threshold =</span> feature.selection.p.threshold,</span>
<span id="cb5-87"><a href="#cb5-87"></a>        <span class="dt">training.datasets =</span> training.datasets,</span>
<span id="cb5-88"><a href="#cb5-88"></a>        <span class="dt">validation.datasets =</span> validation.datasets,</span>
<span id="cb5-89"><a href="#cb5-89"></a>        <span class="dt">learning.algorithms =</span> learning.algorithms,</span>
<span id="cb5-90"><a href="#cb5-90"></a>        <span class="dt">top.n.features =</span> top.n</span>
<span id="cb5-91"><a href="#cb5-91"></a>        );</span>
<span id="cb5-92"><a href="#cb5-92"></a></span>
<span id="cb5-93"><a href="#cb5-93"></a>    <span class="co"># perform Kaplan-Meier analysis and generate plots</span></span>
<span id="cb5-94"><a href="#cb5-94"></a>    <span class="kw">create.survivalplots</span>(</span>
<span id="cb5-95"><a href="#cb5-95"></a>        <span class="dt">data.directory =</span> data.directory,</span>
<span id="cb5-96"><a href="#cb5-96"></a>        <span class="dt">output.directory =</span> output.directory,</span>
<span id="cb5-97"><a href="#cb5-97"></a>        <span class="dt">training.datasets =</span> training.datasets,</span>
<span id="cb5-98"><a href="#cb5-98"></a>        <span class="dt">validation.datasets =</span> validation.datasets,</span>
<span id="cb5-99"><a href="#cb5-99"></a>        <span class="dt">top.n.features =</span> top.n,</span>
<span id="cb5-100"><a href="#cb5-100"></a>        <span class="dt">learning.algorithms =</span> learning.algorithms,</span>
<span id="cb5-101"><a href="#cb5-101"></a>        <span class="dt">truncate.survival =</span> truncate.survival,</span>
<span id="cb5-102"><a href="#cb5-102"></a>        <span class="dt">survtime.cutoffs =</span> <span class="kw">c</span>(<span class="dv">5</span>),</span>
<span id="cb5-103"><a href="#cb5-103"></a>        <span class="dt">main.title =</span> <span class="ot">FALSE</span>,</span>
<span id="cb5-104"><a href="#cb5-104"></a>        <span class="dt">KM.plotting.fun =</span> <span class="st">&quot;create.KM.plot&quot;</span>,</span>
<span id="cb5-105"><a href="#cb5-105"></a>        <span class="dt">resolution =</span> <span class="dv">100</span></span>
<span id="cb5-106"><a href="#cb5-106"></a>        );</span>
<span id="cb5-107"><a href="#cb5-107"></a>    }</span></code></pre></div>
<p>Please note that a number of parameters such as <code>output.directory, training.datasets and validations.datasets</code> are repeatedly passed in various methods. This may look mindless, however, this is because none of the methods return objects that are passed over to the next method/s. The rationale behind this was to keep the memory footprint to bare-minimum by avoiding large R objects passed around. This particular feature also facilitates restarting from any step after deliberate or accidental closure of R environment as everything can be read again from the filesystem. The only compromise is the data footprint on the filesystem.</p>
<p>The above R-script will generate two sub-directories under the <code>output.directory</code> path:</p>
<blockquote>
<p>output/ and graphs/</p>
</blockquote>
</div>
<div id="output" class="section level2">
<h2><a id="Output"></a> Output</h2>
<p>SIMMS creates two output directories; <code>output/ and graphs/</code>. The contents of these directories are described below:</p>
<div id="output-1" class="section level3">
<h3>output/</h3>
<blockquote>
<p>coxph_nodes__<code>$training.datasets</code>__datatype_<code>$data.types</code>.txt coxph_edges_coef__<code>$training.datasets</code>__datatype_<code>$data.types</code>.txt coxph_edges_P__<code>$training.datasets</code>__datatype_<code>$data.types</code>.txt</p>
</blockquote>
<p>Contains per feature (nodes) univariate Cox proportional hazards model results, and per interaction (gene-gene edge) Cox proportional hazards model results</p>
<blockquote>
<p>top_subnets_score__TRAINING_<code>$training.datasets</code>__model_<code>1,2,3</code>__PV_<code>$feature.selection.p.thresholds</code>.txt</p>
</blockquote>
<p>Contains per subnetwork scores and is used for subsequent feature selection. Models 1, 2 and 3 refers to Node+Interaction, Node-only and Interaction-only models. Nodel-only (Model-2) is generally the most interpretable and robust model</p>
<blockquote>
<p>riskscores_uv__<code>all_training,all_validation</code>__TRAINING_<code>$training.datasets</code>__model_<code>1,2,3</code>__top_<code>$top.n</code>.txt</p>
</blockquote>
<p>Contains per sample risk scores for each subnetwork, along with Survival time and Survival status. Sample by risk score matrix. First two columns contain survival data</p>
<blockquote>
<p>riskgroups_uv__<code>all_training,all_validation</code>__TRAINING_<code>$training.datasets</code>__model_<code>1,2,3</code>__top_<code>$top.n</code>.txt</p>
</blockquote>
<p>Contains per sample risk groups for each subnetwork, along with Survival time and Survival status. Sample by risk group matrix. First two columns contain survival data. Risk groups are median-dichotomised (training set) risk scores</p>
<blockquote>
<p>riskscores__<code>all_training,all_validation</code>__TRAINING_<code>$training.datasets</code>__<code>backward,forward,glm,randomforest</code>__model_<code>1,2,3</code>__top_<code>$top.n</code>.txt</p>
</blockquote>
<p>Contains per sample risk scores estimated by the multivariate Cox proportional hazards model, along with Survival time and Survival status</p>
<blockquote>
<p>riskgroups__<code>all_training,all_validation</code>__TRAINING_<code>$training.datasets</code>__<code>backward,forward,glm,randomforest</code>__model_<code>1,2,3</code>__top_<code>$top.n</code>.txt</p>
</blockquote>
<p>Contains per sample risk group (along with Survival time and Survival status) generated by median dichotomising risk scores (training set) estimated through the multivariate Cox proportional hazards model</p>
<blockquote>
<p>coxph__<code>all_training,all_validation</code>__TRAINING_<code>$training.datasets</code>__<code>backward,forward,glm,randomforest</code>__model_<code>1,2,3</code>__top_<code>$top.n</code>.txt</p>
</blockquote>
<p>Contains univariate Cox proportional hazards model results with risk group (derived from multivariate Cox model) as the explanatory variable</p>
<blockquote>
<p>beta__TRAINING_<code>$training.datasets</code>__<code>backward,forward</code>__model_<code>1,2,3</code>__top_<code>$top.n</code>.txt</p>
</blockquote>
<p>Contains the fitted coefficients (betas) of the final model following backward elimination and forward selection (separately)</p>
<blockquote>
<p>beta__TRAINING_<code>$training.datasets</code>__<code>glm</code>__model_<code>1,2,3</code>__top_<code>$top.n</code>.txt</p>
</blockquote>
<p>Contains the fitted coefficients (betas) of the final model following GLMs (LASSO, Ridge or Elastic Nets) driven feature selection</p>
<blockquote>
<p>vimp__TRAINING_<code>$training.datasets</code>__<code>randomforest</code>__model_<code>1,2,3</code>__top_<code>$top.n</code>.txt</p>
</blockquote>
<p>Contains the variable importance of random forest.</p>
<blockquote>
<p>OOB_error__TRAINING_<code>$training.datasets</code>__<code>randomforest</code>__model_<code>1,2,3</code>__top_<code>$top.n</code>.txt</p>
</blockquote>
<p>Contains OOB error rate against number of trees to help identify stablisation point for ‘rf.ntree’ parameter</p>
</div>
<div id="graphs" class="section level3">
<h3>graphs/</h3>
<blockquote>
<p>KM_uv__<code>all_training,all_validation</code>__TRAINING_<code>$training.datasets</code>__model_<code>1,2,3</code>__<code>SubnetworkName</code>__truncated_<code>$truncate.survival</code>.png</p>
</blockquote>
<p>Kaplan-Meier analysis of risk groups generated for each subnetwork through univariate Cox proportional hazards model. These will be generated if parameter <code>plot.univariate.data</code> was set to <code>TRUE</code> in <code>create.survivalplots()</code> as default value is set to <code>FALSE</code>.</p>
<blockquote>
<p>KM__<code>all_training,all_validation</code>__TRAINING_<code>$traning.datasets</code>__<code>backward,forward,glm,randomforest</code>__model_<code>1,2,3</code>__top_<code>$top.n</code>__truncated_<code>$truncate.survival</code>.png</p>
</blockquote>
<p>Kaplan-Meier analysis of risk groups generated through multivariate Cox proportional hazards model</p>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
